src/frontend/apps/lti_site/components/App/AppContentLoader/index.tsx

```diff
@@ -104,8 +90,43 @@ const AppContentLoader = () => {
     useCurrentUser.setState({
       currentUser,
     });
+  }, [decodedJwt]);
+
+  return <Content />;
+};
 
+const AppContentLoader = () => {
+  const [isLoaded, setIsLoaded] = useState(false);
+  const queryClient = new QueryClient();
+
+  const defaultResourceContext: ResourceContext = {
+    permissions: {
+      can_access_dashboard: false,
+      can_update: false,
+    },
+    resource_id: '',
+    roles: [],
+  };
+
+  //  load it from the store to prevent having a dependancy and recompute decodedJwt
+  const decodedJwt = useMemo(() => {
+    const jwt = useJwt.getState().jwt;
+    if (jwt) {
+      setIsLoaded(true);
+      return decodeJwt(jwt);
```

Comment:
Here you coul probably use `return useJwt.getState().getDecodedJwt();`. Doing this, the jet id decoded once and then store to reuse it later.